contrib/kktix_bq_etl.py

```diff
@@ -0,0 +1,559 @@
+    if args.ticket_year:    
+        ticket_year = args.ticket_year            
+
+    # Set up the client object
+    client = bigquery.Client(project=project_id)
+
+    # Set up the job config
+    job_config = bigquery.QueryJobConfig()
+    job_config.use_legacy_sql = False
+
+    # Build the SQL query to extract the data
+    query = f'SELECT * FROM `{TABLE}` WHERE lower( NAME ) LIKE "%{ticket_year}%{ticket_type}%"'
+
+    # Execute the query and extract the data 
+    # bigquery depends on db_dtypes for to_dataframe(), depends on apache-arrow, cannot be installed (build failed) on my old MAC
+    # use the raw result from bigQuery is good for this use case
+    job = client.query(query, job_config=job_config)
+    results = job.result() 
+
+    # print(results)
+
+    print(f'Extracted from {TABLE}  successful.')
+  
+    # Use DataFrame for table transform operations
+    df = pd.DataFrame()
+    for row in results:
+        json_dict = json.loads(row['ATTENDEE_INFO'])
+        df_dict = pd.DataFrame([json_dict])
+
+        # We don't have paid_date column from ATTENDEE_INFO, convert the updated_at timestamp
+        timestamp = df_dict.loc[0, 'updated_at']
+        # print(timestamp)
+        dt_object = datetime.fromtimestamp(timestamp)
+        transc_date = dt_object.strftime('%Y-%m-%d')
+        # print(transc_date)    
+        df_dict['paid_date'] = transc_date
+
+        # ticket_type value is always "qrcode" here, and conflict with old table, 
+        # kyc, id_number, slot value is always empty
+        # 'id','ticket_id','state','checkin_code', 'qrcode', updated_at, order_no no exist in old table
+        # data will be handle later
+        useless_columns = ['id','ticket_id','state','checkin_code', 'qrcode', 'currency', 'data', 'ticket_type', 'kyc', 'updated_at', 'id_number', 'slot', 'order_no']
+        df_dict = df_dict.drop(columns = useless_columns) 
+        df_dict = df_dict.rename(columns = {"reg_no": "registration_no", "ticket_name": "ticket_type", "is_paid": "payment_status"})
+        if 'payment_status' in df_dict.columns:
+            if df_dict.loc[0, 'payment_status']:
+                # df_dict['payment_status'] = df_dict['payment_status'].astype("string")
+                df_dict.loc[0, 'payment_status'] = "paid"
+            else:
+                df_dict.loc[0, 'payment_status'] = "not"    
+        # json_dict['data'] in format of [[item_name, item_value],[],...]
+        data_list = json_dict['data'] 
+        column_list = [dl[0] for dl in data_list]
+        value_list = [dl[1] for dl in data_list]
+        # print(column_list)
+        # print(value_list)
+        df_data = pd.DataFrame()
+        df_data[column_list] = [value_list]
+
+        df_dict = pd.concat([df_dict, df_data], axis=1)
+        df = pd.concat([df, df_dict], ignore_index=True)
+
+    # print(df.columns)
+    sanitized_df = sanitize_column_names(df)
+    hash_privacy_info(sanitized_df)    
```

Comment:
thx for remembering hashing these sensitive columns. However, we've already encrypted data on `extract` side. so no need to do it again~
https://github.com/pycontw/pycon-etl/blob/810096e2b8bbd711bd8e98b99744cd16356bdf50/dags/ods/kktix_ticket_orders/udfs/kktix_api.py#L33

Reply:
yes, could be removed after confirmed.